src/spec-node/singleContainer.ts

```diff
@@ -126,26 +125,35 @@ async function buildAndExtendImage(buildParams: DockerResolverParameters, config
 	const { config } = configWithRaw;
 	const dockerfileUri = getDockerfilePath(cliHost, config);
 	const dockerfilePath = await uriToWSLFsPath(dockerfileUri, cliHost);
+	const buildContextPath = await uriToWSLFsPath(getDockerContextPath(cliHost, config), cliHost)
 	if (!cliHost.isFile(dockerfilePath)) {
 		throw new ContainerError({ description: `Dockerfile (${dockerfilePath}) not found.` });
 	}
 
 	let dockerfile = (await cliHost.readFile(dockerfilePath)).toString();
 	const originalDockerfile = dockerfile;
 	let baseName = 'dev_container_auto_added_stage_label';
-	if (config.build?.target) {
-		// Explictly set build target for the dev container build features on that
-		baseName = config.build.target;
-	} else {
-		// Use the last stage in the Dockerfile
-		// Find the last line that starts with "FROM" (possibly preceeded by white-space)
-		const { lastStageName, modifiedDockerfile } = ensureDockerfileHasFinalStageName(dockerfile, baseName);
-		baseName = lastStageName;
-		if (modifiedDockerfile) {
-			dockerfile = modifiedDockerfile;
+
+	const intermediateBuildAargs = ["build", "-f", dockerfilePath, "-t"];
```

Comment:
Using two builds had the issues mentioned in https://github.com/microsoft/vscode-remote-release/issues/6848#issuecomment-1316691017.